src/scenic/simulators/metadrive/simulator.py

```
@@ -0,0 +1,321 @@
+
+    def createSimulation(self, scene, *, timestep, **kwargs):
+        if timestep is not None and timestep != self.timestep:
+            raise RuntimeError(
+                "cannot customize timestep for individual MetaDrive simulations; "
+                "set timestep when creating the MetaDriveSimulator instead"
+            )
+        self.scenario_number += 1
+        return MetaDriveSimulation(
+            scene,
+            render=self.render,
+            render3D=self.render3D,
+            scenario_number=self.scenario_number,
+            timestep=self.timestep,
+            sumo_map=self.sumo_map,
+            real_time=self.real_time,
+            center_x=self.center_x,
+            center_y=self.center_y,
+            offset_x=self.offset_x,
+            offset_y=self.offset_y,
+            sumo_map_boundary=self.sumo_map_boundary,
+            **kwargs,
+        )
+
+    def destroy(self):
+        super().destroy()
+
+
+class MetaDriveSimulation(DrivingSimulation):
+    def __init__(
+        self,
+        scene,
+        render,
+        render3D,
+        scenario_number,
+        timestep,
+        sumo_map,
+        real_time,
+        center_x,
+        center_y,
+        offset_x,
+        offset_y,
+        sumo_map_boundary,
+        **kwargs,
+    ):
+        # NOTE: MetaDrive requires at least one agent to be defined per simulation run,
+        # and the first object must be a car (to serve as the ego vehicle).
+        try:
+            if len(scene.objects) == 0:
+                raise SimulationCreationError(
+                    "Metadrive requires you to define at least one Scenic object."
+                )
+            if not scene.objects[0].isCar:
+                raise SimulationCreationError(
+                    "The first object must be a car to serve as the ego vehicle in Metadrive."
+                )
+        except SimulationCreationError as e:
+            print(e)
+            sys.exit(1)
```

Was this debugging code left in by accident? If it's intentional, I think raising and then immediately catching an exception is confusing: I'd just do what you want directly. But we shouldn't just call `sys.exit` in case the user wants to catch this and handle it at a higher level. If `SimulationCreationError` isn't a good exception here since the bug in the user's scenario (no ego) is guaranteed to cause a failure every time, you could raise `InvalidScenarioError`.

---

Fixed as suggested in commit 89878f32b7a4dcb66c54d287524193c0d030c1be